added ending, tidied up

evanmcook opened the pull request:
added functionality in parts.lua, where a random number between  450 and 500 inclusive is decided as the ending measure number.
added graphics methods in order to reformat redraw() to adhere to style. added resetSong() but i am unsure if resetSong() is needed, as i have assumed the user will retrigger the song using k3, and then k2.

evanmcook committed:
added ending, tidied up

added functionality in parts.lua, where a random number between  450 and 500 inclusive is decided as the ending measure number.
added graphics methods in order to reformat redraw() to adhere to style. added resetSong() but i am unsure if resetSong() is needed, as i have assumed the user will retrigger the song using k3, and then k2.

tyleretters commented on lib/parts.lua:
```diff
@@ -108,7 +111,11 @@ function parts:whole_notes(phase)
+  elseif song.measure == 24 then self.drumPat = 4 
+  elseif song.measure >= parts.endingMeas then  
+    song.is_screen_dirty = true
+    song.is_playing = false 
```
i'd recommend swapping these two lines for two reasons:
1. it will make it clear to future developers (probably you!) that setting `is_playing` to false then requires a redraw. as is, that is not apparent.
2. i'm not a lua expert, but i think there is a chance that the redraw will happen *between* these two lines... meaning that the redraw will happen, the the song will stop playing.

tyleretters replied:
did you mean to add `resetSong()` here?

evanmcook replied:
> i'd recommend swapping these two lines for two reasons:
> 
>     1. it will make it clear to future developers (probably you!) that setting `is_playing` to false then requires a redraw. as is, that is not apparent.
> 
>     2. i'm not a lua expert, but i think there is a chance that the redraw will happen _between_ these two lines... meaning that the redraw will happen, the the song will stop playing.

ooh, great catch. thank you.

> did you mean to add `resetSong()` here?

for the 'thank you' message to be readable without disappearing unexpectedly, i've only put `resetSong()` in `key()` and added instructions on how to start the song again in the 'thank you' message.

tyleretters commented on song.lua:
```diff
@@ -104,6 +78,13 @@ function enc(e, d)
 end
 
+
+function resetSong()
```
well done!

evanmcook committed:
Update parts.lua

fixed order of lines to make sure the song stops playing first, and then the screen is marked as dirty
